VirtoCommerce.InventoryModule.Web/Controllers/Api/InventoryModuleController.cs

```diff
@@ -89,42 +102,31 @@ public IHttpActionResult DeleteFulfillmentCenters([FromUri] string[] ids)
         /// </summary>
         /// <remarks>Get inventory of products for each fulfillment center.</remarks>
         /// <param name="ids">Products ids</param>
+        /// <param name="fulfillmentCenterIds">The fulfillment centers that will be used to filter product inventories</param>
 		[HttpGet]
         [Route("products")]
         [ResponseType(typeof(InventoryInfo[]))]
-        public IHttpActionResult GetProductsInventories([FromUri] string[] ids)
+        public IHttpActionResult GetProductsInventories([FromUri] string[] ids, string[] fulfillmentCenterIds = null)
         {
-            var result = new List<InventoryInfo>();
-            var allFulfillments = _fulfillmentCenterSearchService.SearchCenters(new FulfillmentCenterSearchCriteria { Take = int.MaxValue }).Results;
-            var inventories = _inventoryService.GetProductsInventoryInfos(ids);
-            foreach (var productId in ids)
-            {
-                foreach (var fulfillment in allFulfillments)
-                {
-                    var inventory = inventories.FirstOrDefault(x => x.ProductId == productId && x.FulfillmentCenterId == fulfillment.Id);
-                    if(inventory == null)
-                    {
-                        inventory = AbstractTypeFactory<InventoryInfo>.TryCreateInstance();
-                        inventory.ProductId = productId;
-                        inventory.FulfillmentCenter = fulfillment;
-                        inventory.FulfillmentCenterId = fulfillment.Id;
-                    }
-                    result.Add(inventory);
-                }
-            }
+            var criteria = AbstractTypeFactory<InventorySearchCriteria>.TryCreateInstance();
+            criteria.FulfillmentCenterIds = fulfillmentCenterIds;
+            criteria.ProductIds = ids;
 
-            return Ok(result.ToArray());
+            var result = _inventorySearchService.SearchInventories(criteria);
```

Comment:
This will return only 20 records

Also you should format the document automatically when saving the document.